docs/standards/dynamic-inventory-pattern.md

```
@@ -0,0 +1,206 @@
+      {%- endfor -%}
+      {{ nodes }}
+```
+
+### Using DNS Instead of IPs
+
+```yaml
+# Preferred when possible - use DNS names
+vars:
+  database_host: "db.example.com"        # DNS name - passes validation
+  api_endpoint: "api.service.consul"     # Service discovery - passes validation
+```
```

markdownlint-cli2 (0.17.2)

Lists should be surrounded by blank lines

(MD032, blanks-around-lists)